Add unstable `rustc-check-cfg` build script output (#10539)
rust-lang/cargo: src/doc/src/reference/unstable.md

cargo check -Z unstable-options -Z check-cfg=features,names,values
```

Or for `output`:

```rust,no_run
// build.rs
println!("cargo:rustc-check-cfg=names(foo, bar)");
```

```
cargo check -Z unstable-options -Z check-cfg=output
```

### `cargo:rustc-check-cfg=CHECK_CFG`

The `rustc-check-cfg` instruction tells Cargo to pass the given value to the
`--check-cfg` flag to the compiler. This may be used for compile-time
detection of unexpected conditional compilation name and/or values.

> This can only be used is combination with `-Zcheck-cfg=output` otherwise it's just a nop.

Reviewer: ```suggestion
This can only be used in combination with `-Zcheck-cfg=output` otherwise it is ignored with a warning.
```
